File: 0942.增减字符串匹配/0942-增减字符串匹配.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int n = s.length();
        vector<int> ans(n+1);
        int l = 0;
        int r = n;

        for( int i=0; i<n; i++ ) {
            char c = s[i];
            if( c == 'I' ) {
                ans[i] = l;;
                l++;
            } else {
                ans[i] = r;
                r--;
            }
        }

        ans[n] = l;

        return ans;
    }
};

File: 0942.增减字符串匹配/0942-增减字符串匹配.go
func diStringMatch(s string) []int {
	n := len(s)
	ans := make([]int, n+1)
	l := 0
	r := n

	for i, ch := range s {
		if ch == 'I' {
			ans[i] = l
			l++
		} else {
			ans[i] = r
			r--
		}
	}

	ans[n] = l
	return ans
}
